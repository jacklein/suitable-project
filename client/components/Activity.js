import React from 'react';
import PropTypes from 'prop-types';
import { Text, StyleSheet } from 'react-native'
import { Card, Button } from 'react-native-elements';
import { primaryColor } from '../styles/common';

const Activity = ({ activity, onPress }) => {
  const { details } = activity;
  return (
    <Card containerStyle={styles.card} >
      <Text style={styles.title}>
        {details.title}
      </Text>
      <Text style={styles.description}>
        {details.description}
      </Text>
      <Button
        icon={{ name:'check', color:'#ffffff' }}
        backgroundColor={primaryColor}
        buttonStyle={styles.button}
        title='Mark Complete'
        onPress={() => onPress(activity)}
      />
    </Card>
  );
}

const styles = StyleSheet.create({
  card: {
    borderWidth: 0
  },
  title: {
    textAlign: 'center',
    fontWeight: 'bold', 
    marginBottom: 15
  },
  description: {
    marginBottom: 20
  },
  button: {
    borderRadius: 0,
    alignSelf: 'center'
  }
});

Activity.propTypes = {
  activity: PropTypes.object,
  onPress: PropTypes.func
}


export default Activity;