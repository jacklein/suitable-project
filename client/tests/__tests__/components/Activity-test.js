import React from 'react';
import { shallow } from 'enzyme';
import Activity from '../../../components/Activity';
import { activity } from '../../mockdata';

describe('Activity component', () => {
  test('renders correctly', () => {
    const wrapper = shallow(
      <Activity activity={activity} onPress={() => {}}/>
    );
    expect(wrapper).toMatchSnapshot();
  });
});