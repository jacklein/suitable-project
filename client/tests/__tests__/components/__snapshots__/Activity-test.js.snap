// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Activity component renders correctly 1`] = `
<Card
  containerStyle={
    Object {
      "borderWidth": 0,
      "marginBottom": 1,
      "marginTop": 10,
    }
  }
>
  <Component
    style={
      Object {
        "fontWeight": "bold",
        "marginBottom": 10,
      }
    }
  >
    Bacon 21151
  </Component>
  <Component
    style={
      Object {
        "marginBottom": 15,
      }
    }
  >
    Fuga id aspernatur sed pariatur itaque placeat. Velit unde aut explicabo. Pariatur fuga eveniet ipsa. Aut amet repudiandae laboriosam doloremque sit. Ut quam molestiae ab occaecati illo sed unde. Harum minus est libero eaque cum.
  </Component>
  <Button
    backgroundColor="#4492E5"
    buttonStyle={
      Object {
        "borderRadius": 0,
        "marginBottom": 15,
      }
    }
    icon={
      Object {
        "color": "#ffffff",
        "name": "check",
      }
    }
    onPress={[Function]}
    title="Mark Activity Complete"
  />
</Card>
`;
